src/main/java/apoc/bolt/Bolt.java

```diff
@@ -0,0 +1,160 @@
+        Node node = internalValue.asNode();
+        if (virtual) {
+            VirtualNode virtualNode = new VirtualNode(node.id(), db);
+            node.labels().forEach(l -> virtualNode.addLabel(Label.label(l)));
+            node.asMap().entrySet().iterator().forEachRemaining(i -> virtualNode.setProperty(i.getKey(), i.getValue()));
+            return Util.map(key, virtualNode);
+        } else
+            return Util.map(key, Util.map("entityType", internalValue.type().name(), "labels", node.labels(), "id", node.id(), "properties", node.asMap()));
+    }
+
+    private Map<String, Object> toRelationship(Object value, Map<String, Object> map, String key, boolean virtual) {
+        Value internalValue = ((InternalEntity) value).asValue();
+        Relationship relationship = internalValue.asRelationship();
+        if (virtual) {
+            VirtualNode start = new VirtualNode(relationship.startNodeId(), db);
+            VirtualNode end = new VirtualNode(relationship.endNodeId(), db);
+            if (map.values().contains(start) && map.values().contains(end))
+                return map;
+            VirtualRelationship virtualRelationship = new VirtualRelationship(start, end, RelationshipType.withName(relationship.type()));
+            relationship.asMap().entrySet().iterator().forEachRemaining(i -> virtualRelationship.setProperty(i.getKey(), i.getValue()));
+            return Util.map(key, virtualRelationship);
+        } else
+            return Util.map(key, Util.map("entityType", internalValue.type().name(), "type", relationship.type(), "id", relationship.id(), "start", relationship.startNodeId(), "end", relationship.endNodeId(), "properties", relationship.asMap()));
+    }
+
+    private Map<String, Object> toPath(Object value, boolean virtual, String key) {
+        Map<String, Object> map = new HashMap<>();
+        InternalPath value1 = (InternalPath) value;
+        Path path = value1.asValue().asPath();
+        path.spliterator().forEachRemaining(segment -> {
+            if (virtual) {
+                VirtualNode startNode = new VirtualNode(segment.start().id(), db);
+                segment.start().labels().forEach(l -> startNode.addLabel(Label.label(l)));
+                segment.start().asMap().entrySet().iterator().forEachRemaining(p -> startNode.setProperty(p.getKey(), p.getValue()));
+                VirtualNode endNode = new VirtualNode(segment.end().id(), db);
+                segment.end().labels().forEach(l -> endNode.addLabel(Label.label(l)));
+                segment.end().asMap().entrySet().iterator().forEachRemaining(p -> endNode.setProperty(p.getKey(), p.getValue()));
+                map.put(key, Util.map("start", startNode, "end", endNode));
+            } else
+                map.put(key, Util.map("entityType", "NODE", "startLabels", segment.start().labels(), "startId", segment.start().id(), "type", segment.relationship().type(), "endLabels", segment.end().labels(),
```

Comment:
I don't think this works. as you are overrding the same entry time and again.
it should be a path entry with entityTypePath and a collection of "node" and collection of "relationships"